contracts/MarketContract.sol

```diff
@@ -72,20 +79,36 @@ contract MarketContract is Creatable {
 
         PRICE_DECIMAL_PLACES = contractSpecs[2];
         QTY_MULTIPLIER = contractSpecs[3];
-        EXPIRATION = contractSpecs[4];
-        require(EXPIRATION > now);
+        EXPIRATION = contractSpecs[5];
+        require(EXPIRATION > now, 'expiration must be in the future');
 
-        CONTRACT_NAME = contractName;
-        creator = baseAddresses[0];
         COLLATERAL_TOKEN_ADDRESS = baseAddresses[1];
         COLLATERAL_POOL_ADDRESS = baseAddresses[2];
         COLLATERAL_PER_UNIT = MathLib.calculateTotalCollateral(PRICE_FLOOR, PRICE_CAP, QTY_MULTIPLIER);
+        FEE_PER_UNIT = MathLib.calculateFeePerUnit(PRICE_FLOOR, PRICE_CAP, QTY_MULTIPLIER, contractSpecs[4]);
+
+        // create long and short tokens
+        StringLib.slice memory pathSlice = contractNames.toSlice();
+        StringLib.slice memory delim = ",".toSlice();
+        require(pathSlice.count(delim) == 2, "ContractNames must contain 3 names");  //contractName,lTokenName,sTokenName
+        CONTRACT_NAME = pathSlice.split(delim).toString();
+
+        PositionToken longPosToken = new PositionToken("Long Position Token", pathSlice.split(delim).toString(), 0);
```

Comment:
I got confused in the whole pathSlice logic, we should probably add a test to MarketContracts to ensure the correct symbol is extracted for its positions tokens symbol.

What I want to ask here though is it okay to leave `Long Position Token` and `Short Position Token` as the name for all MPX Position tokens (symbols are different alright)?

Reply:
Also, it seems contract names have changed from what we have before, we should confirm in our other codebase that we are no more parsing names based on the old convention.

Reply:
I think so, I am debating making it `MARKET Protocol Long Position Token` so people know where it came from as well.